_posts/deployment/2017-08-04-deploy-from-source.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Deploy Directly from Source
+modified_at: 2017-08-04 00:00:00
+category: deployment
+tags: source deploy
+order: 1
+---
+
+Deploy directly from source is available on Scalingo. Indeed it is possible to deploy an archive which contains your application directly from your shell.
+
+Before using this type of deployment, you need an auth-token for Scalingo that you can find on your [dashboard](https://my.scalingo.com/profile) or from our API. Every information are [here](https://developers.scalingo.com/index.html#authentication).
+
+You also need to have created an application on our platform with [my.scalingo.com](https://my.scalingo.com) or [from your shell](https://developers.scalingo.com/apps.html#create-an-application).
+
+## The Workflow
+
+The first step of your deployment is to archive your application code in the format `tar.gz`.
+
+After that, you have to push your code on a `source` provide by Scalingo (see the section Create a Source) with an `upload_url` (which is explain in the section Update your Application).
+
+The application is now able to be deploy thanks to the `download_url` (see the section Deploy your application).
+
+
+## Create a Source
+
+To create a source you will have to make a POST request on `https://api.scalingo.com/v1/sources`.
+
+```bash
+curl -H "Accept: application/json" -H "Content-Type: application/json" -u :$AUTH_TOKEN \
+  -X POST https://api.scalingo.com/v1/sources -d ''
+```
+
+The respond will look like:
+
+```json
+{
+  "upload_url": "https://api.scalingo.com/v1/sources/123e4567-e89b-12d3-a456-426655440000?token=dc958153c3cd32659ffad5deeda9405d",
+  "download_url": "https://api.scalingo.com/v1/sources/123e4567-e89b-12d3-a456-426655440000?token=9df650a60014571abff0ee4e2d06a8fc"
+}
+```
+
+`upload_url`: used to upload your application on the source. 
```

Comment:
I still can't understand both explanations about upload_url and download_url. When reading this page, you never explain what a "source" is.